packages/core/src/internal/runSaga.js

```diff
@@ -85,11 +86,14 @@ export function runSaga(options, saga, ...args) {
     finalizeRunEffect,
   }
 
+  suspend()
   const task = proc(env, iterator, context, effectId, getMetaInfo(saga), null)
```

Comment:
Could we move `suspend()` and `flush()` into `proc` as follows?

```javascript
export default function proc(env, iterator, parentContext, parentEffectId, meta, cont) {
  suspend()
  // other code...
  flush()
  return task
```

If we do this, we just need write `suspend()` and `flush()` **one time**. Not sure this idea would work, but tests are passing in my PC.

Reply:
Further more, if we do like this, we may export a function `wrap` (or some other better name) from *packages/core/src/internal/scheduler.js* and use it as a decorator on `proc()`.

```javascript
export function wrap(fn) {
  return (...args) => {
    try {
      suspend()
      return fn(...args)
    } finally {
      flush()
    }
  }
}
```

Reply:
would prefer avoiding additional closures in the saga code, so Im reluctant to using decorator approach

and Im not sure if we can move this safely into the proc, as `runForkEffect` would *slightly* change the order of how things happen (when the cb callback gets invoked) - I will experiment with it though

Reply:
It seems that tests are passing if we move this to proc, I'll try to do further experimentation to see if I can break some case